### PR TITLE
feat: Update Subscription data model to support anniversary date

### PR DESCRIPTION
**Pull request by vincent-pochet**
## Context

We want to add the ability to choose to bill users at subscription date anniversary and not only on a calendar basis.

## Objective

The objective of this changes is to prepare the introduction of the new billing time:

- Add `billing_time` enum to subscription
  - Possible values: `anniversary` / `calendar`
  - Default value: `calendar` to keep it compatible with existing subscription / API

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -19,11 +19,10 @@
   create_table "active_storage_attachments", id: :uuid, default: -> { "gen_random_uuid()" }, force: :cascade do |t|
     t.string "name", null: false
     t.string "record_type", null: false
-    t.uuid "record_id", null: false
     t.uuid "blob_id", null: false
     t.datetime "created_at", null: false
+    t.uuid "record_id"
```
Comment (rsempe) [inline db/schema.rb:24]: Some changes are not related to this PR, is that expected?

File: db/schema.rb [hunk 2]
```diff
@@ -344,10 +344,10 @@
...
     t.string "currency", null: false
     t.string "name"
-    t.string "rate_amount", null: false
-    t.string "credits_balance", default: "0.00", null: false
-    t.string "balance", default: "0.00", null: false
-    t.string "consumed_credits", default: "0.00", null: false
+    t.decimal "rate_amount", precision: 5, default: "0", null: false
+    t.decimal "credits_balance", precision: 5, default: "0", null: false
+    t.decimal "balance", precision: 5, default: "0", null: false
+    t.decimal "consumed_credits", precision: 5, default: "0", null: false
```
Comment (rsempe) [inline db/schema.rb:350]: Same here

**Review by rsempe — APPROVED** [review]

File: db/schema.rb (hunk 2 above)
Comment (vincent-pochet) [reply db/schema.rb:350]: :thinking: weird...

I'll try to fix it
